tests/backup/backup_test.go

```diff
@@ -86,6 +88,508 @@ var _ = Describe("{UserGroupManagement}", func() {
+
+		Step("Verify if 'test app spec'", func() {
+
+			log.InfoD("Allowed apps are %v", appsWithCRDsAndWebhooks)
+			for i := 0; i < len(appList); i++ {
+				contains := Contains(appsWithCRDsAndWebhooks, appList[i])
+				dash.VerifyFatal(contains, true,
+					fmt.Sprintf("check if app [%s] can be used as a dummy spec for *this* test", appList[i]))
+			}
+		})
+
+		Step("Verify K8s version of Src and Dest Cluster", func() {
+			log.InfoD("Verify K8s version of Src and Dest Cluster")
+
+			Step("Register cluster for backup", func() {
+				log.InfoD("Register cluster for backup")
+				ctx, err := backup.GetAdminCtxFromSecret()
+				log.FailOnError(err, "Fetching px-central-admin ctx")
+				err = CreateSourceAndDestClusters(orgID, "", "", ctx)
+				dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+				clusterStatus, clusterUid = Inst().Backup.RegisterBackupCluster(orgID, SourceClusterName, "")
+				dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, "Verifying backup cluster")
+			})
+
+			Step("Switch Context (\"Source\")", func() {
+				sourceClusterConfigPath, err := GetSourceClusterConfigPath()
+				log.FailOnError(err, "Failed to get kubeconfig path for source cluster. Error: [%v]", err)
+
+				err = Inst().S.SetConfig(sourceClusterConfigPath)
+				log.FailOnError(err, "Failed to switch to context to source cluster [%v]", sourceClusterConfigPath)
+
+				srcMaj, srcMin, err = k8s.ClusterVersion()
+				log.FailOnError(err, "Failed to get Source Cluster Version")
+			})
+
+			Step("Switch Context (\"destination\")", func() {
+				destinationClusterConfigPath, err := GetDestinationClusterConfigPath()
+				log.FailOnError(err, "Failed to get kubeconfig path for destination cluster. Error: [%v]", err)
+
+				err = Inst().S.SetConfig(destinationClusterConfigPath)
+				log.FailOnError(err, "Failed to switch to context to destination cluster [%v]", destinationClusterConfigPath)
+
+				destMaj, destMin, err = k8s.ClusterVersion()
+				log.FailOnError(err, "Failed to get Destination Cluster Version")
+			})
+
+			Step("Compare Source and Destination cluster version numbers", func() {
+
+				if srcMaj != 0 && destMaj != 0 {
+
+					log.InfoD("Source Cluster version: %d.%d ; Destination Cluster version: %d.%d",
+						srcMaj, srcMin, destMaj, destMin)
+
+					const multiple = 100
+					isValid := (destMaj*multiple + destMin) >= (srcMaj*multiple + srcMin)
+
+					dash.VerifyFatal(isValid, true,
+						"This test is only meant for cases where the Source Cluster version is LESSER than the Destination Cluster version.")
+
+				} else {
+					err := fmt.Errorf("Cannot compare Source and Destination CLuster versions due to invalid data in Source Cluster version (%d.%d), Destination Cluster (%d.%d)", srcMaj, srcMin, destMaj, destMin)
+					log.FailOnError(err, "Failed to validate K8s versions")
+				}
+
+			})
```

Comment:
Combine all these steps into 1 single step using the semver library as suggested.
And more importantly, the version check of source and destination cluster is a pre-requisite for the test case not the actual part of the test case. So it would be more appropriate if you move this under `JustBeforeEach `

Reply:
While I'll combine these steps using semver library, I don't think I can/should move this to `JustBeforeEach`, as the cluster version checks can only be executed **after** `CreateSourceAndDestClusters`, which is the process of registering the clusters, which, at least the way it's done in most other tests, is part of the test itself.